### ISSUE TITLE
Support multiple instances

### ISSUE THREAD

**Issue by chilltemp** [alexziskind1/nativescript-oauth#55]
Right now index.ts has a bunch of functions/variables that are static.  This prevents us from having multiple instances of oauth at once.  My problem is that it makes unit testing harder, but what if someone was coding to multiple auth providers?  (I know it would be better for a single provider with federation to others.  This is just a what-if.)

**Comment by alexziskind1** [comment]
Definitely. This is something I’ve been wanting to do for a little while
now, but haven’t had a chance yet.

On Sep 22, 2017, at 2:31 PM, chilltemp <notifications@github.com> wrote:

Right now index.ts has a bunch of functions/variables that are static. This
prevents us from having multiple instances of oauth at once. My problem is
that it makes unit testing harder, but what if someone was coding to
multiple auth providers? (I know it would be better for a single provider
with federation to others. This is just a what-if.)

—
You are receiving this because you are subscribed to this thread.
Reply to this email directly, view it on GitHub
<https://github.com/alexziskind1/nativescript-oauth/issues/55>, or mute the
thread
